modules/core/src/passes/layers-pass.ts

```diff
@@ -263,7 +265,12 @@ export default class LayersPass extends Pass {
         this._lastRenderIndex = Math.max(this._lastRenderIndex, layerRenderIndex);
 
         // overwrite layer.context.viewport with the sub viewport
-        moduleParameters.viewport = viewport;
+        for (const moduleName in moduleParameters) {
+          const moduleProps = moduleParameters[moduleName];
+          if (moduleProps && typeof moduleProps === 'object' && 'viewport' in moduleProps) {
```

Comment:
Seems there might be some useful props naming conventions implicit in this that perhaps could be adopted by luma?